### PR TITLE
Camel-k-runtime tests is failing in Quarkus platform #5319

### PR DESCRIPTION
**Pull request by spatnity**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by spatnity** [comment]
#5319 Can be fixed when Camel Quarkus 3.5.0-SNAPSHOT is released.

**Comment by aldettinger** [comment]
It's a shame we don't have a IOUtils like one liner to save a resource hierarchy to files. At least, nothing come on top of mind.

File: integration-tests/camel-k-runtime/src/test/java/org/apache/camel/quarkus/k/it/RuntimeTest.java [hunk 1]
```diff
@@ -48,6 +61,51 @@ public void inspect() {
...
+                Files.copy(stream, destination, StandardCopyOption.REPLACE_EXISTING);
+            } catch (Exception e) {
+                throw new RuntimeException("Failed to copy resource " + resourceName + " to temporary directory", e);
+            }
+        }
+
+        @Override
+        public void stop() {
+            try {
+                Files.walk(TEMP_DIR)
```
Comment (aldettinger) [inline integration-tests/camel-k-runtime/src/test/java/org/apache/camel/quarkus/k/it/RuntimeTest.java:99]: Those kinds of delete, we can maybe write less code ala:
https://github.com/apache/camel-quarkus/blob/main/integration-tests/file/src/test/java/org/apache/camel/quarkus/component/file/it/NonFlakyFileTestResource.java#L54

Comment (aldettinger) [reply integration-tests/camel-k-runtime/src/test/java/org/apache/camel/quarkus/k/it/RuntimeTest.java:99]: it's not blocking to merge by the way

**Review by aldettinger — APPROVED** [review]

File: integration-tests/camel-k-runtime/src/test/java/org/apache/camel/quarkus/k/it/RuntimeTest.java (hunk 1 above)
Comment (spatnity) [reply integration-tests/camel-k-runtime/src/test/java/org/apache/camel/quarkus/k/it/RuntimeTest.java:99]: Oh okay, noted. Thankyou @aldettinger

Comment (jamesnetherton) [reply integration-tests/camel-k-runtime/src/test/java/org/apache/camel/quarkus/k/it/RuntimeTest.java:99]: Let's follow up with another PR for this if needed.

**Review by jamesnetherton — APPROVED** [review]

**Pull request merged**